### PR TITLE
feat(onboarding): Persist skipping onboarding

### LINKED ISSUE #1685: [Feature]: Persist skipping the Onboarding Wizard
### What problem does this solve?

If you skip onboarding, Sparky Fitness will re-open the wizard every time you load the page. Before we rework the wizard, it should be possible for skipping of the wizard to persist.

### Describe the feature you'd like

Record that the user skipped the wizard in the database and don't pop it open every time. Add a widget in an obvious place, like the dashboard, to resume the wizard that stays up so long as the wizard hasn't been completed.

### Which area does this relate to?

Web frontend

### How important is this to you?

Would significantly improve my workflow

### Alternatives you've considered

- Completing the wizard, but I'm not sure I understand yet how that will impact my data/use of the app.
- Reworking the wizard so it's clearer, but that will take more time and discussion.

### Additional context

I've almost got  a PR ready for this.

### PR DESCRIPTION
**Pull request by ikogan**
## Description

**What problem does this PR solve?**
Solves #1685 by persisting the fact that onboarding has been skipped to the database and adding a "Complete Setup" button to the top of the screen.

**How did you implement the solution?**

- Added a new `onboarding_skipped` boolean to the `onboarding_status` table.
- Added a new route that flips the default `FALSE` to `TRUE` and mapped the `Skip` button to call that route.
- Changed the way the wizard chooses to display to base it on `onboardingSkipped` _and_ `onboardingComplete`.
- Added a new button that calls `onStartOnboarding`.

> [!NOTE]
> It may be good for someone besides me who has completed onboarding to test and make sure I didn't miss some corner case.

Linked Issue: Closes #1685 

## How to Test

1. Run this branch.
2. Login with a new user or any user who has not completed onboarding.
3. Click "Skip"
4. Reload the page, observe that the onboarding wizard does not open.
5. Click the "Complete Setup" button at the top and observe that the wizard opens.

## PR Type

- [ ] Issue (bug fix)
- [x] New Feature
- [ ] Refactor
- [ ] Documentation

## Checklist

**All PRs:**

- [x] **[MANDATORY - ALL] Integrity & License**: I certify this is my own work, free of malicious code, and I agree to the [License terms](https://github.com/CodeWithCJ/SparkyFitness/blob/main/LICENSE).

**New features only:**

- [x] **[MANDATORY for new feature] Alignment**: I have raised a GitHub issue and it was reviewed/approved by maintainers or it was approved on Discord.

**Frontend changes (`SparkyFitnessFrontend/`):**

- [x] **[MANDATORY for Frontend changes] Quality**: I have run `pnpm run validate` and it passes.
- [x] **[MANDATORY for Frontend changes] Translations**: I have only updated the English (`en`) translation file.

**Backend changes (`SparkyFitnessServer/`):**

- [x] **[MANDATORY for Backend changes] Code Quality**: I have run typecheck, lint, and tests. New files use TypeScript, new endpoints have Zod schemas, and new endpoints include tests.
- [x] **[MANDATORY for Backend changes] Database Security**: I have updated `rls_policies.sql` for any new user-specific tables.

**UI changes (components, screens, pages):**

- [x] **[MANDATORY for UI changes] Screenshots**: I have attached Before/After screenshots below.

**Mobile changes (`SparkyFitnessMobile/`):**

- [ ] **[MANDATORY for Mobile changes] Tested on device or emulator**: I have verified the changes work on iOS or Android.

## Screenshots

<details>
<summary>Click to expand</summary>

### Before

The button in after was missing. Unfortunately that would be challenging to generate now.

### After

Screenshot From 2026-06-30 22-27-34

</details>

## Notes for Reviewers

@CodeWithCJ mentioned that many things won't work or may break if they skip early in the setup. If I recall, I was able to configure those things in the settings myself after skipping the wizard. I think the "Complete Setup" may be sufficiently obvious if they run into issues but I'm open to doing it a different way.

### CODE REVIEW

File: SparkyFitnessServer/models/onboardingRepository.ts [hunk 1]
```diff
@@ -108,11 +109,35 @@ async function resetOnboardingStatus(userId: any) {
...
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function setOnboardingSkipped(userId: any) {
+  const client = await getClient(userId);
+  try {
+    const query = `
+      UPDATE onboarding_status
+      SET onboarding_skipped = TRUE, updated_at = NOW()
+      WHERE user_id = $1;
+    `;
+    await client.query(query, [userId]);
```
[the remark this answers is not in the document]
Comment (ikogan) [reply SparkyFitnessServer/models/onboardingRepository.ts:126]: @CodeWithCJ should this be fixed? I feel a little nervous inadvertently creating rows here since I'm not 100% certain when these are actually supposed to be created.

File: SparkyFitnessServer/config/swagger.ts [hunk 2]
```diff
@@ -1127,6 +1127,7 @@ const options = {
           type: 'object',
           properties: {
             onboarding_complete: { type: 'boolean' },
+            onboarding_skipped: { type: 'boolean' },
           },
```
[the remark this answers is not in the document]
Comment (ikogan) [reply SparkyFitnessServer/config/swagger.ts:1131]: @CodeWithCJ it feels like a lot of the swagger spec documents the responses with snake case but several routes actually return camel case. Is this intended? I feel like I should do whatever the rest of the code is doing even if it's technically incorrect?

**Review by CodeWithCJ — APPROVED** [review]

**Comment by CodeWithCJ** [comment]
@ikogan some times gemini creates false positive comments. we can ignore them for now.
